tests/TestCase/TestSuite/IntegrationTestTraitTest.php

```diff
@@ -513,6 +513,59 @@ public function testPostDataHttpServer()
         $this->assertHeader('X-Middleware', 'true');
     }
 
+    /**
+     * Test that the uploaded files are passed correctly to the request
+     *
+     * @return void
+     */
+    public function testUploadedFiles()
+    {
+        $this->configRequest([
+            'files' => [
+                'file' => [
+                    'tmp_name' => __FILE__,
+                    'size' => 42,
+                    'error' => 0,
+                    'type' => 'text/plain',
+                    'name' => 'Uploaded file'
+                ],
+                'pictures' => [
+                    'name' => [
+                        ['file' => 'a-file.png'],
+                        ['file' => 'a-moose.png']
+                    ],
+                    'type' => [
+                        ['file' => 'image/png'],
+                        ['file' => 'image/jpg']
+                    ],
+                    'tmp_name' => [
+                        ['file' => __FILE__],
+                        ['file' => __FILE__]
+                    ],
+                    'error' => [
+                        ['file' => 0],
+                        ['file' => 0]
+                    ],
+                    'size' => [
+                        ['file' => 17188],
+                        ['file' => 2010]
+                    ],
+                ],
+                'upload' => new UploadedFile(__FILE__, 42, 0)
```

Comment:
This seems like something that should be possible.  Is making this work complicated?